### PR TITLE
FLINK-2018 Add ParameterUtil.fromGenericOptionsParser()

### PR DESCRIPTION
**Pull request by ajaybhat**
Pls review and comment

### CODE REVIEW

**Comment by mbalassi** [comment]
Thanks for picking up the issue. Generally looks good to me. One minor comment: please do not merge my commit on the delta policies as it is already in. :)

**Comment by ajaybhat** [comment]
> please do not merge my commit on the delta policies as it is already in.

I don't know how that happened. As usual I just committed, rebased from master and pushed to my fork. Is there a way to get rid of that commit?

File: flink-java/src/test/java/org/apache/flink/api/java/utils/ParameterToolTest.java [hunk 1]
```diff
@@ -150,6 +150,14 @@ public void testMerged() {
 		validate(parameter);
 	}
 
+	@Test
+	public void testFromGenericOptionsParser() throws IOException {
+		System.setProperty("input", "myInput");
+		System.setProperty("expectedCount", "15");
+		ParameterTool parameter = ParameterTool.fromGenericOptionsParser(new String[]{"-D", "input=myInput", "-DexpectedCount=15"});
```
Comment (rmetzger) [inline flink-java/src/test/java/org/apache/flink/api/java/utils/ParameterToolTest.java]: Is the genericOptionsParser getting the settings from the system properties or the command line arguments?

Comment (ajaybhat) [inline flink-java/src/test/java/org/apache/flink/api/java/utils/ParameterToolTest.java]: Its from the command line arguments.

Comment (rmetzger) [inline flink-java/src/test/java/org/apache/flink/api/java/utils/ParameterToolTest.java]: Then why are you setting the two system properties?

Comment (ajaybhat) [inline flink-java/src/test/java/org/apache/flink/api/java/utils/ParameterToolTest.java]: It was a mistake; I'll remove those.

File: flink-java/src/test/java/org/apache/flink/api/java/utils/ParameterToolTest.java [hunk 2]
```diff
@@ -150,6 +150,12 @@ public void testMerged() {
 		validate(parameter);
 	}
 
+	@Test
+	public void testFromGenericOptionsParser() throws IOException {
+		ParameterTool parameter = ParameterTool.fromGenericOptionsParser(new String[]{"-D", "input=myInput", "-DexpectedCount=15"});
+		validate(parameter);
+	}
```
Comment (rmetzger) [inline flink-java/src/test/java/org/apache/flink/api/java/utils/ParameterToolTest.java]: Sorry that I didn't write this earlier when I did the first review of the pull request:
Can you also add a test where we read one of the arguments (say input=myinput) from a xml configuration file.
As far as I can see, the GenericOptionsParser also supports specifying `-conf path/to/file.xml`.

That would be amazing ,)

**Comment by fhueske** [comment]
Github sometimes includes unrelated commits into PRs when the Apache Git repository and the Github mirror are out-of-sync, i.e., commits from the Apache Git have not been mirrored yet.

@ajaybhat The commit should disappear when you do another rebase and push --force into your PR branch.

**Comment by StephanEwen** [comment]
Looks good. I will merge this...

**Pull request closed without merging**